src/app/(post-login)/submission/components/ProgramMenu.tsx

```diff
@@ -54,27 +65,29 @@ const StatusMenuItem: FC<{ children: ReactNode }> = ({ children }) => {
 const ProgramMenu = ({ shortNameSearchQuery }: { shortNameSearchQuery: string }) => {
 	const params = useParams();
 	const pathname = usePathname();
-	const router = useRouter();
+	const { egoJwt } = useAuthContext();
+	const { DATA_CENTER } = useAppConfigContext();
 
 	// params can be arrays from dynamic routing
 	const activeProgramName = typeof params.shortName === 'string' ? params.shortName : '';
 
-	const { data: programsData, loading, error } = useQuery(SIDEMENU_PROGRAMS);
+	const {
+		data: programsData,
+		loading,
+		error,
+	} = useQuery(SIDEMENU_PROGRAMS, { variables: { dataCenter: DATA_CENTER } });
 	const programs = programsData?.programs || [];
 
 	const filteredPrograms = !shortNameSearchQuery.length
-		? programs
-		: programs.filter(
-				(program) =>
-					program && program.shortName.search(new RegExp(shortNameSearchQuery, 'i')) > -1,
-		  );
-
-	const setActiveProgram =
-		(shortName: string): MouseEventHandler =>
-		() => {
-			const url = `/submission/program/${shortName}/registration`;
-			router.push(url);
-		};
+		? programs.filter(notNull)
+		: programs
+				.filter(notNull)
+				.filter((program) => program.shortName.search(new RegExp(shortNameSearchQuery, 'i')) > -1);
```

Comment:
I'm not sure the `.filter(notNull)` is needed. the gql types are currently broke but this can be investigated

Reply:
potentially just change line 81 to :
```javascript
	const programs = programsData?.programs?.filter(notNull) || [];
```

Reply:
as we're filtering by `notNull` either way

Reply:
Nice, much improved